StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentMethodType.swift

```diff
@@ -126,6 +126,19 @@ extension PaymentSheet {
                 }
             }
 
+            if
+                recommendedStripePaymentMethodTypes.contains(.link),
+                !recommendedStripePaymentMethodTypes.contains(.USBankAccount),
+                !intent.isDeferredIntent,
+                intent.linkFundingSources?.contains(.bankAccount) == true
+            {
+                // we must add this BEFORE we do filtering via
+                // `PaymentSheet.PaymentMethodType.supportsAdding`
+                // because we want to filter out instant debits
+                // IF the user has not linked Financial Connections
+                recommendedStripePaymentMethodTypes.append(.instantDebits)
+            }
```

Comment:
I moved the code up so we could start hitting the `recommendedStripePaymentMethodTypes.filter { paymentMethodType` code

We want to execute that code so we could check whether Financial Connections is linked:

```
                    case .instantDebits:
                        return [.financialConnectionsSDK]
```

Reply:
This is the Android equivalent (note that thier ".link" check is more upstream so its not immediately obvious in this snippet):

https://github.com/stripe/stripe-android/blob/9785cc59af2773ec3bd059ce7ab168080060ce08/paymentsheet/src/main/java/com/stripe/android/lpmfoundations/paymentmethod/AddPaymentMethodRequirement.kt#L60-L68

Reply:
It's surprising to me that `instantDebits` is added despite it not being, apparently, on the payment method types. Like, it's valid to confirm a PI with `instantDebits` payment method type even if the PI.payment_method_types does not contain `instantDebits`?

Reply:
Wow, interesting!

Reply:
No I don't think it is, its just for PaymentSheet client-side code (the switch I mentioned earlier happens client-side)

Reply:
Ah okay. We should remove it from STPPaymentMethodType then, that is a (public) type that corresponds to https://docs.stripe.com/api/payment_methods/object#payment_method_object-type.

Reply:
hopefully we can do it as part of a separate PR as this particular decision was approved in https://github.com/stripe/stripe-ios/pull/3528 (happy to refactor [will just need help from payment on how else to structure it])

Reply:
Yes please!